site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -15201,7 +15202,19 @@ public void doUnjoin(RunData data) {
 		
 		// refresh the whole page
 		scheduleTopRefresh();
```

Comment:
I think you should drop the scheduleTopRefresh as otherwise you'll get a double reload when unjoining and staying in the same site.